src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -735,18 +581,20 @@ async Task<bool> IsUpToDateInternalAsync(CancellationToken token)
 
                 try
                 {
-                    if (!CheckGlobalConditions(buildAction, logger))
+                    State state = _state;
```

Comment:
Here we copy a reference to the state and pass it around on the stack, so it doesn't matter if Dataflow triggers an update to the state while we're calculating the result here.